### ISSUE TITLE
Allow code using AWS SDK for Go v2 to be built in CodeBuild

### ISSUE THREAD

**Issue by frankoid** [aws/aws-sdk-go-v2#1014]
**Is your feature request related to a problem? Please describe.**
AWS SDK for Go v2 needs Go 1.15 according to https://aws.amazon.com/blogs/developer/aws-sdk-for-go-version-2-v2-release-candidate/
However, the CodeBuild standard docker images do not support Go 1.15 yet, see https://github.com/aws/aws-codebuild-docker-images/issues/377 and https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html

We build our lambdas in CodeBuild using the standard images so the latest Go version we can use is 1.14 which according to the documentation prevents us from using the AWS SDK for Go v2.

**Describe the solution you'd like**
Get the CodeBuild team to release support for Go 1.15.

**Describe alternatives you've considered**
Support Go 1.14 in AWS SDK for Go v2.

**Additional context**
N/A

**Comment by jasdel** [comment]
Thanks for reaching out @frankoid the best way to configure a AWS CodeBuild project to use Go 1.15 is to use a custom `Environment` configuration.

* `Environment image` as `Custom Image`
* `Image registry` as `Other registry`
* `External registry URL` as `golang:1.15` 

![Screen Shot 2021-01-05 at 10 36 10](https://user-images.githubusercontent.com/961963/103685486-2c06a180-4f42-11eb-9dab-ff3f8a157668.png)

An alternative to using the public `golang:1.15` image is to use a custom container stored in a container repository, such as using [Amazon ECR](https://aws.amazon.com/ecr/)

Benefit of using either the public `golang` or custom container repository is that you can directly control the version of Go your applications are built with without waiting for AWS CodeBuild to update their build in images.

**Comment by frankoid** [comment]
Thanks for the suggestion, and the quick response. Yes a custom image would work however there are a couple of downsides, one that I think would affect anyone using the custom image approach and one that is more specific to us:

1. The CodeBuild provisioning step is faster when using a recent version of a standard CodeBuild image because CodeBuild keeps these images around on build hosts so doesn't have to pull them over the network. I'm pretty sure I've seen this documented somewhere but I can't find a link right now, sorry. I've observed it in practice though, provisioning for one of our builds started to take a few minutes because we were using an old, no-longer-kept-locally image and speeded up to less than a minute once we upgraded to the latest.
2. We spin up another Docker container inside our CodeBuild build to run DynamoDB-local in for our tests to run against. I expect that golang:1.15 doesn't have docker support so we'd have to build our own custom image with Go 1.15 + Docker support. Totally doable, but extra effort.

So in summary, there are ways for us to use AWS SDK for Go v2 right now, but the slower build times and extra effort would probably be enough to put me off trying out the SDK v2 for the time being.

Once again, thanks for the suggestion, I appreciate you taking the time to respond 👍

**Comment by jasdel** [comment]
Thanks for the update and additional clarification on the issue you're running into. Reaching out to CodeBuild via their [AWS Forums](https://forums.aws.amazon.com/forum.jspa?forumID=230&start=0) is also a good way to make this suggest to support Go 1.15 directly in CodeBuild curated environment images.

**Comment by frankoid** [comment]
Yes I've [already posted on the CodeBuild forum](https://forums.aws.amazon.com/thread.jspa?threadID=333462&tstart=0), no response as yet.

In isolation I think it's reasonable to require Go 1.15 at this stage (it's been out a few months) but it's a shame that these two AWS products don't play well together out of the box, I was hoping raising this issue might prompt some discussions between the SDK and CodeBuild teams.

**Comment by jasdel** [comment]
I wanted to let you know that CodeBuild now supports Go 1.15 in its `Ubuntu standard:5.0` image.

**Comment by frankoid** [comment]
Thanks @jasdel, we are using that image now.
